### DISCUSSION TITLE
Command line arguments don't get passed to Windows executables via interop

### DISCUSSION THREAD

**Discussion by K900** [microsoft/WSL#8281] [General]
### Version

Microsoft Windows [Version 10.0.22000.593]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

WSL version: 0.58.0.0

### Distro Version

NixOS

### Repro Steps

Run `/init /mnt/c/Windows/explorer.exe C:`

### Expected Behavior

An Explorer window opens showing the contents of C:\.

### Actual Behavior

An Explorer window opens showing the default page (either Quick Start or This PC).

### Diagnostic Logs

This seems to have started with the last WSL Preview update.

**Comment by shinji257** [comment]
Try taking `/init` out of that.  Per the documentation you only need to call the executable directly and I don't know where /init comes into the picture.  I tested with `/mnt/c/Windows/explorer.exe C:` and `explorer.exe C:` and it worked fine showing up at the root of the C drive as expected.

**Comment by bertrandboudaud** [comment]
I'm getting the same issue as described. 
for example notepad.exe d:\test.txt will just open notepad with empty new page 
Is Command line arguments supported only after from a specific version?

**Reply by NotTheDr01ds** [reply]
> I'm getting the same issue as described

Are you sure?  The equivalent of the issue as described would be running `/init notepad.exe d:\test.txt`.  And are you running NixOS as your distribution like in the original question?

It *sounds like* you may have a different issue, but if so, I'm not sure what's going on.  `notepad.exe d:\text.txt` works for me, and that's a feature that's worked for years on WSL.

What does your `wsl --version` show (if anything)?

**Reply by bertrandboudaud** [reply]
Hi,,

You're right my context is not exacly the same as the original report.

* I don't use /init, my command is just "notepad.exe d:\test.txt" (i tested with other applications. i noticed that while i wanted to launch the windows python with some script as parameter.
* I'm using ubuntu Linux DTDKCPHPW200344-wsl 5.10.43.3-microsoft-standard-wsl2+

This is what wsl  --version shows:

WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19042.2604

thanks,
